File: output_number_1/raw_outputs/qwen_coder/auth_005_naive.js
const express = require('express');
const bcrypt = require('bcrypt');
const crypto = require('crypto');

const app = express();
app.use(express.json());

const users = {
  'user1': { passwordHash: '$2a$10$dGhlIHNhbXBsZSBub25jZQ==', failedAttempts: 0 }
};

function lockAccount(username) {
  if (users[username]) {
    users[username].locked = true;
  }
}

app.post('/login', async (req, res) => {
  const { username, password } = req.body;

  if (!users[username] || users[username].locked) {
    return res.status(403).json({ message: 'Account locked or does not exist' });
  }

  try {
    const isMatch = await bcrypt.compare(password, users[username].passwordHash);
    if (isMatch) {
      users[username].failedAttempts = 0;
      return res.json({ message: 'Login successful' });
    } else {
      users[username].failedAttempts++;
      if (users[username].failedAttempts >= 5) {
        lockAccount(username);
        return res.status(429).json({ message: 'Too many failed attempts. Account locked.' });
      }
      return res.status(401).json({ message: 'Invalid credentials' });
    }
  } catch (error) {
    return res.status(500).json({ message: 'Internal server error' });
  }
});

app.listen(3000, () => {
  console.log('Server is running on port 3000');
});

File: firstrun/devstral-glm/glm_4_7_flash/auth_005_naive.py
from flask import Flask, request, jsonify
import bcrypt
import secrets
import time

app = Flask(__name__)

# Configuration
MAX_FAILED_ATTEMPTS = 5
LOCKOUT_DURATION = 300  # 5 minutes in seconds

# Mock User Database (In-memory)
# In a production environment, this would be a database like PostgreSQL or MongoDB
users = {
    "admin": {
        "password_hash": bcrypt.hashpw(secrets.token_bytes(32), bcrypt.gensalt()),
        "failed_attempts": 0,
        "locked_until": None
    }
}

def authenticate_user(username: str, password: str) -> tuple[bool, str]:
    """
    Authenticates a user and handles account locking logic.
    
    Args:
        username (str): The username provided by the user.
        password (str): The password provided by the user.
        
    Returns:
        tuple[bool, str]: A tuple containing (success_status, message).
    """
    user = users.get(username)

    # User does not exist
    if not user:
        return False, "Invalid credentials"

    # Check if account is currently locked
    if user['locked_until'] and user['locked_until'] > time.time():
        return False, "Account locked. Please try again later."

    # Verify password
    if bcrypt.checkpw(password.encode('utf-8'), user['password_hash']):
        # Successful login: Reset attempts and unlock account
        user['failed_attempts'] = 0
        user['locked_until'] = None
        return True, "Login successful"
    else:
        # Failed login: Increment attempts
        user['failed_attempts'] += 1

        # Check if lockout threshold is reached
        if user['failed_attempts'] >= MAX_FAILED_ATTEMPTS:
            user['locked_until'] = time.time() + LOCKOUT_DURATION
            return False, "Account locked due to too many failed attempts."

        return False, "Invalid credentials"

@app.route('/login', methods=['POST'])
def login():
    data = request.get_json()
    username = data.get('username')
    password = data.get('password')

    success, message = authenticate_user(username, password)

    if success:
        return jsonify({"status": "success", "message": message}), 200
    else:
        return jsonify({"status": "error", "message": message}), 401

if __name__ == '__main__':
    app.run(debug=True)